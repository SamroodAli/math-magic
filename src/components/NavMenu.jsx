import { Menu } from 'semantic-ui-react';
import { Link } from 'react-router-dom';

const NavMenu = ({ activeItem, handleItemClick }) => (
  <Menu fluid vertical tabular>
    <Menu.Item
      as={Link}
      to="/"
      name="bio"
      active={activeItem === 'bio'}
      onClick={handleItemClick}
    />
    <Menu.Item
      as={Link}
      to="/calculator"
      name="calculator"
      active={activeItem === 'calculator'}
      onClick={handleItemClick}
    />
    <Menu.Item
      as={Link}
      to="/quotes"
      name="quotes"
      active={activeItem === 'quotes'}
      onClick={handleItemClick}
    />
  </Menu>
);
