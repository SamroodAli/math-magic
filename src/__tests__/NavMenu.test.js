import renderer from 'react-test-renderer';
import { StaticRouter } from 'react-router-dom';
import NavMenu from '../components/NavMenu';

test('NavMenu displays all items as shown in snapshot', () => {
  const tree = renderer.create(<StaticRouter><NavMenu activeItem="Home" handleItemClick={jest.fn()} /></StaticRouter>).toJSON();
  expect(tree).toMatchSnapshot();
});
