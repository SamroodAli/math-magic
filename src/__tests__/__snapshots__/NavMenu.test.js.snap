// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavMenu displays all items as shown in snapshot 1`] = `
<div
  className="ui fluid vertical tabular menu"
>
  <a
    className="active item"
    href="/"
    onClick={[Function]}
  >
    Bio
  </a>
  <a
    className="item"
    href="/calculator"
    onClick={[Function]}
  >
    Calculator
  </a>
  <a
    className="item"
    href="/quotes"
    onClick={[Function]}
  >
    Quotes
  </a>
</div>
`;
